tests/test/mod.rs

```diff
@@ -1,300 +1,12 @@
-            println!("----------------------------------------------------------------------");
-            println!("goal {}", goal_text);
-            assert!(goal_text.starts_with("{"));
-            assert!(goal_text.ends_with("}"));
-            let goal = chalk_parse::parse_goal(&goal_text[1..goal_text.len() - 1])
-                .unwrap()
-                .lower(&*program)
-                .unwrap();
-
-            println!("using solver: {:?}", solver_choice);
-            let peeled_goal = goal.into_peeled_goal(db.interner());
-            match expected {
-                TestGoal::Aggregated(expected) => {
-                    let result = db.solve(&peeled_goal);
-                    assert_result(result, expected);
-                }
-                TestGoal::All(expected) => {
-                    let mut expected = expected.into_iter();
-                    assert!(
-                        db.solve_multiple(&peeled_goal, |result, next_result| {
-                            match expected.next() {
-                                Some(expected) => {
-                                    assert_same(
-                                        &format!(
-                                            "{}",
-                                            result.as_ref().map(|v| v.display(&ChalkIr))
-                                        ),
-                                        expected,
-                                    );
-                                }
-                                None => {
-                                    assert!(!next_result, "Unexpected next solution");
-                                }
-                            }
-                            true
-                        }),
-                        "Not all solutions processed"
-                    );
-                    if expected.next().is_some() {
-                        panic!("Not all solutions processed");
-                    }
-                }
-                TestGoal::First(expected) => {
-                    let mut expected = expected.into_iter();
-                    db.solve_multiple(&peeled_goal, |result, next_result| match expected.next() {
-                        Some(solution) => {
-                            assert_same(
-                                &format!("{}", result.as_ref().map(|v| v.display(&ChalkIr))),
-                                solution,
-                            );
-                            if !next_result {
-                                assert!(expected.next().is_none(), "Not enough solutions found");
-                            }
-                            true
-                        }
-                        None => false,
-                    });
-                    assert!(expected.next().is_none(), "Not enough solutions found");
-                }
-            }
-        });
-    }
-}
+pub use super::test_util::{test::*, *};
```

Comment:
Why move these? I'm not entirely against this change, but I think this PR is big enough that if you want to do this, splitting it into a separate PR would be helpful.

Reply:
Right - I think we wanted to be able to use the internals from `test!` in `logging_db_output_sufficient!` defined `test_util/logging_db.rs`, and I think there were problems using macros from `tests/test/mod.rs` in `tests/test_util/logging_db.rs` with `test/` already using macros from `test_util.rs`.

This change is a separate commit, so making a separate PR wouldn't be hard at all. We could also revert it and work around this some less invasive way - like putting the macro for `logging_db` in `tests/logging_db/mod.rs` and just letting the macro dependency chain be `logging_db -> test -> test_util`.

Reply:
We've removed this change, and can make another PR to do the misc. fixing later.